index.js

```diff
@@ -8,9 +8,58 @@
 import './i18nInit';
 import { AppRegistry } from 'react-native';
 import { setJSExceptionHandler } from 'react-native-exception-handler';
+import messaging from '@react-native-firebase/messaging';
+import notifee, { EventType } from '@notifee/react-native';
 import App from './src/App';
 import { name as appName } from './app.json';
 import { errorHandler } from './src/errorHandler';
+import { messageHandler, setInitialNotificationData } from './src/sagas/pushNotification';
+
+/**
+ * This function is called when the app is in background or **quit** state
+ * and a push notification from **notifee** is received. It also handles
+ * notification events like dismiss and action press.
+ *
+ * @see https://notifee.app/react-native/reference/eventtype
+ *
+ * In this project the Notifee is used to interprete the data message received
+ * from firebase and show the notification to the user.
+ *
+ */
+const setNotifeeBackgroundListener = () => {
```

Comment:
I think I prefer having the `index.js` as clean as possible, so I feel we could move those methods to an utils file and just calling them here. What do you think?